src/Psalm/Internal/Analyzer/AttributesAnalyzer.php

```diff
@@ -0,0 +1,377 @@
+                        )
+                    ]);
+                }
+
+                if (!$first_arg_type->isSingleIntLiteral()) {
+                    return 63; // Fall back to default if it's invalid
+                }
+
+                return $first_arg_type->getSingleIntLiteral()->value;
+            }
+        }
+
+        IssueBuffer::maybeAdd(
+            new InvalidAttribute(
+                "The class {$attribute_name} doesn't have the Attribute attribute",
+                $attribute_location
+            ),
+            $suppressed_issues
+        );
+
+        return 63; // Fall back to default if it's invalid
+    }
+
+    /**
+     * @param iterable<AttributeGroup> $attribute_groups
+     *
+     * @return Generator<int, Attribute>
+     */
+    private static function iterateAttributeNodes(iterable $attribute_groups): Generator
+    {
+        foreach ($attribute_groups as $attribute_group) {
+            foreach ($attribute_group->attrs as $attribute) {
+                yield $attribute;
+            }
+        }
+    }
+
+    /**
+     * Analyze Reflection getAttributes method calls.
+
+     * @param list<Arg> $args
+     */
+    public static function analyzeGetAttributes(
+        StatementsAnalyzer $statements_analyzer,
+        string $method_id,
+        array $args
+    ): void {
+        if (count($args) !== 1) {
+            // We skip this analysis if $flags is specified on getAttributes, since the only option
+            // is ReflectionAttribute::IS_INSTANCEOF, which causes getAttributes to return children.
+            // When returning children we don't want to limit this since a child could add a target.
+            return;
+        }
+
+        switch ($method_id) {
+            case "ReflectionClass::getattributes":
+                $target = 1;
+                break;
+            case "ReflectionFunction::getattributes":
+                $target = 2;
+                break;
+            case "ReflectionMethod::getattributes":
+                $target = 4;
+                break;
+            case "ReflectionProperty::getattributes":
+                $target = 8;
+                break;
+            case "ReflectionClassConstant::getattributes":
+                $target = 16;
+                break;
+            case "ReflectionParameter::getattributes":
+                $target = 32;
+                break;
+            default:
+                return;
+        }
+
+        $arg = $args[0];
+        if ($arg->name !== null) {
+            for (; !empty($args) && ($arg->name->name ?? null) !== "name"; $arg = array_shift($args));
+            if ($arg->name->name ?? null !== "name") {
+                // No named argument for "name" parameter
+                return;
+            }
+        }
+
+        $arg_type = $statements_analyzer->getNodeTypeProvider()->getType($arg->value);
+        if ($arg_type === null || !$arg_type->isSingle() || !$arg_type->hasLiteralString()) {
```

Comment:
Union::getSingleStringLiteral would probably allow you to be more concise here

Reply:
I don't think so, it's possible it's not a single literal and I don't want it to throw an exception in that case.